### PR TITLE
changes to sanitizer arrangement for #676 and #679

### PR DESCRIPTION
**Pull request by 6pac**

### CODE REVIEW

File: slick.grid.js [hunk 1]
```diff
@@ -116,7 +116,9 @@ if (typeof Slick === "undefined") {
       viewportMaxWidthPx: undefined,
       suppressCssChangesOnHiddenInit: false,
       ffMaxSupportedCssHeight: 6000000,
-      maxSupportedCssHeight: 1000000000
+      maxSupportedCssHeight: 1000000000,
+      sanitizer: undefined, // sanitize function, builtin is: defaultSanitizeHtmlString(dirtyHtml)
+      logSanitizedHtml: true // log to console when sanitised
```
Comment (ghiscoding) [inline slick.grid.js:121]: can that be disabled by default?

File: slick.grid.js [hunk 2]
```diff
@@ -5949,12 +5951,24 @@ if (typeof Slick === "undefined") {
...
+       return dirtyHtmlStr.replace(/(\b)(on[a-z]+)(\s*)=|javascript:([^>]*)[^>]*|(<\s*)(\/*)script([<>]*).*(<\s*)(\/*)script(>*)|(&lt;)(\/*)(script|script defer)(.*)(&gt;|&gt;">)/gi, '');
+    }
+
+    function externalSanitizeHtmlString(dirtyHtml) {
+      if (typeof dirtyHtml !== 'string') return dirtyHtml;
+      var sanitizer = options.sanitizer || defaultSanitizeHtmlString;
+      return sanitizer(dirtyHtml);
+    }
+
+    function internalSanitizeHtmlString(dirtyHtml) {
```
Comment (ghiscoding) [inline slick.grid.js:5965]: what's the difference between the internal/external? I would think that 1 type is enough to drive them all across the project

**Comment by 6pac** [comment]
OK, is that better? 

Re the logging, have changed the default to false but would prefer true.
I assume that if something gets sanitised, you firstly really want to know about it (because it's likely something will break as a result), and secondly you'd probably want to change the code being sanitised if you can.
The sanitiser is not part of your normal development cycle - it's a last ditch measure to avoid malicious injection.

So the idea is that when you turn the sanitizer on, you get messages and make any changes you need to (those changes ideally resulting in zero sanitizer changes = zero  logging output). If you really want to suppress logging output, then you change the option to false.

The main problem at the moment is that messages will be per-cell so you might get an awful lot of messages at first. Perhaps it would be worth counting and limiting the number of messages to say 50.

**Comment by ghiscoding** [comment]
yeah sure I'm ok with the log, I can disable it my libs, no problem. 
The rewrite is a lot simpler but I wonder, how can you ever use the `defaultSanitizeHtmlString`? I mean, I don't see any reference to it, how can the user chose to use it then? Wouldn't it be better with a flag to use that default sanitizer or not? `enableDefaultSanitizer` like I sent you in separate mail

**Comment by 6pac** [comment]
updated again

**Comment by ghiscoding** [comment]
that looks good now, I'd be ok to merge. good job

**Comment by 6pac** [comment]
Great. There were a few different points of view but I think the code ended up better for it.

**Comment by ghiscoding** [comment]
@6pac you should probably close the 2 issues that were mentioned

**Comment by 6pac** [comment]
Sure. Just FYI I'm away remote until the end of August now. I'll be online  generally but may be out of contact for 2-3 days at a stretch.

**Pull request merged**